runtime/binding-mqtt/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/internal/config/MqttRouteConfig.java

```diff
@@ -39,7 +39,7 @@ public MqttRouteConfig(
         this.id = route.id;
         this.when = route.when.stream()
             .map(MqttConditionConfig.class::cast)
-            .map(MqttConditionMatcher::new)
+            .map(conf -> new MqttConditionMatcher(conf, route.guarded))
```

Comment:
```suggestion
            .map(c -> new MqttConditionMatcher(this, c))
```
So you can use `route.identity(guard, authorization)` from `MqttConditionMatcher` constructor (see below).

Add `Map<String, LongFunction<String>> identities` private field and populate in constructor.

Add `String identity(String guard, long authorization)` to `MqttRouteConfig`.
```
String identity(
    String guard,
    long authorization)
{
    return identities.getOrDefault(guard, a -> null).apply(authorization);
}
```

Reply:
Done, thanks!

Reply:
I am reopening this because this caused a bug while I was applying the last suggestions to the PR:

When we pass `this` to the `MqttConditionMatcher` constructor, it receives an instance of `MqttRouteConfig` that is not completely initialized, and this might produce unexpected behaviour when accessing uninitialized `MqttRouteConfig` fields inside the `MqttConditionMatcher` constructor.

In this specific case, the `MqttConditionMatcher` constructor creates a `Function` variable and assigns it to `route::identity`, but the `route.identities` field, while being final, is not yet initialized at this time, and this produce a NullPointerException when the `Function.apply()` method is called in the future, complaining that the `route.identities` field is null.

To solve this I moved the `when` field initialization at the bottom of the `MqttRouteConfig` constructor, so that all the other fields are initialized when calling the `MqttConditionMatcher` constructor with `this` as an input. But this still feels kind of wierd, so maybe it is better to pass the `route.identities` field to the `MqttConditionMatcher` constructor, rather than the whole  `MqttRouteConfig` object which is not fully initialized yet? :thinking: 

Let me know!

Reply:
Yes, agreed.

Let's pass `this::identity` instead of `this` and receive as `Function<String, LongFunction<String>> identities` to pass to `TopicMatcher` directly.

Reply:
Ok, done!